src/crypto/tls/handshake_client_tls13.go

```diff
@@ -409,6 +411,7 @@ func (hs *clientHandshakeStateTLS13) processServerHello() error {
 	}
 
 	if !hs.serverHello.selectedIdentityPresent {
+		hs.handshakeTimings.ReadServerHello = hs.handshakeTimings.elapsedTime()
```

Comment:
Good catch. To avoid calling this twice, I think we should add the following to the top of `processServerHello()`:
```
defer func() {
    hs.handshakeTimings.ReadServerHello = hs.handshakeTimings.elapsedTime()
}()
```

Reply:
I'm not sure. We want this to cover the entire cost of processing the SH, and that means running to the end of each branch.

Reply:
Fixed.